qtconsole/tests/test_jupyter_widget.py

```diff
@@ -82,3 +82,28 @@ def test_other_output(self):
             '<p style="-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;"><br /></p>\n'
             '<p style=" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;"><span style=" color:#000080;">In [</span><span style=" font-weight:600; color:#000080;">2</span><span style=" color:#000080;">]:</span> </p></body></html>'
         ))
+
+    def test_copy_paste(self):
+        """Test copy/paste removes partial and full prompts."""
+        w = JupyterWidget(kind='rich')
+        w._show_interpreter_prompt(1)
+        control = w._control
+
+        code = "    if True:\n        print('a')"
+        w._set_input_buffer(code)
+        assert code not in control.toPlainText()
+
+        cursor = w._get_prompt_cursor()
+
+        pos = cursor.position()
+        cursor.setPosition(pos - 3)
+        cursor.movePosition(QtGui.QTextCursor.End,
+                            QtGui.QTextCursor.KeepAnchor)
+        control.setTextCursor(cursor)
+        control.hasFocus = lambda: True
+        w.copy()
+        clipboard = QtWidgets.QApplication.clipboard()
+        assert clipboard.text() == code
+        w.paste()
+        expected = "In [1]: if True:\n   ...:     print('a')"
```

Comment:
I don't understand this. The text is copied without prompts but when pasted it does show a prompt?

Reply:
Expected is the content of the console. You can't replace the existing prompt when pasting, and a continuation prompt must be created when pasting multilines

Reply:
Ok, I see, thanks for the clarification.